tests/prof_tests.c

```diff
@@ -1,96 +1,1339 @@
 #include "catch.hpp"
 #include "../code/prof.h"
 #include "../code/merc.h"
+#include "../code/db.h"
+#include "test_helpers.c"
 
-SCENARIO("testing interpreting proficiency commands", "[InterpCommand]")
+SCENARIO("performing an index lookup based on proficiency name", "[ProfIndexLookup]")
+{
+	GIVEN("an uninitialized parameter")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+
+			THEN("it should return -1")
+			{
+				auto result = player->Profs()->ProfIndexLookup(nullptr);
+
+				REQUIRE(result == -1);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+
+	GIVEN("an invalid proficiency name")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+
+			THEN("it should return -1")
+			{
+				auto result = player->Profs()->ProfIndexLookup("ÿßã");
+
+				REQUIRE(result == -1);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+
+	GIVEN("an incomplete proficiency name")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+
+			THEN("it should return -1")
+			{
+				auto result = player->Profs()->ProfIndexLookup("cook");
+
+				REQUIRE(result == -1);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+
+	GIVEN("a valid proficiency name")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+
+			THEN("it should return a valid index")
+			{
+				auto result = player->Profs()->ProfIndexLookup("cooking");
+
+				REQUIRE(result == 6);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+}
+
+SCENARIO("check to see if a character has a proficiency (int)", "[HasProf]")
+{
+	GIVEN("an invalid index")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+
+			THEN("it should return false")
+			{
+				auto result = player->Profs()->HasProf(-1);
+
+				REQUIRE(result == false);
+			}
+
+			THEN("it should return false")
+			{
+				auto result = player->Profs()->HasProf(MAX_PROFS + 1);
+
+				REQUIRE(result == false);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+
+	GIVEN("a character who doesn't have the proficiency")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+
+			THEN("it should return false")
+			{
+				auto result = player->Profs()->HasProf(6);
+
+				REQUIRE(result == false);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+
+	GIVEN("a character who doesn't have the proficiency")
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+			player->Profs()->SetProf(6, 1); // cooking
+
+			THEN("it should return false")
```

Comment:
Checking for true in the body but says false

Reply:
copypasta strikes again